crypto/src/key/mod.rs

```diff
@@ -102,13 +104,29 @@ impl PrivateKey {
     }
 }
 
+impl Derivable for PrivateKey {
```

Comment:
The problem with this assumption is that we're saying that *every* kind of PrivateKey now, and in the future, must support being derivable. Not sure this is a good assumption, or even a tenable one.

Reply:
That is correct. The assumption is that because `derive_child` is fallible it would not create any issues.

Another approach would be to have a new private key type `ExtendedPrivateKey` that behaves like a `PrivateKey`.